src/lane_follow/include/LineDetect.h

```diff
@@ -0,0 +1,194 @@
+/*------------------------------------------------------------------------------------------*\
+Lane Detection
+
+General idea and some code modified from:
+chapter 7 of Computer Vision Programming using the OpenCV Library.
+by Robert Laganiere, Packt Publishing, 2011.
+
+ This program is free software; permission is hereby granted to use, copy, modify,
+and distribute this source code, or portions thereof, for any purpose, without fee,
+subject to the restriction that the copyright notice may not be removed
+or altered from any source or altered source distribution.
+The software is released on an as-is basis and without any warranties of any kind.
+In particular, the software is not guaranteed to be fault-tolerant or free from failure.
+The author disclaims all warranties with regard to this software, any use,
+and any consequent failure, is purely the responsibility of the user.
+ 
+Copyright (C) 2013 Jason Dorweiler, www.transistor.io
+\*------------------------------------------------------------------------------------------*/
+
+/*
+ * Created by: Raad Khan
+ * Created On: April 23, 2017
+ * Description:
+ * Usage:
+ * Subscribes to:
+ * Publishes to:
+ *
+ */
+
+#ifndef LINEF
+#define LINEF
+
+#include <ros/ros.h>
+#include <ros/package.h>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <ros/console.h>
+#include <sb_utils.h>
+#include <ros/time.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include "opencv2/highgui/highgui.hpp"
+#include <opencv2/objdetect/objdetect.hpp>
+#include <iostream>
+#include <vector>
+#include <stdio.h>

```

Comment:
We want to minimize the number of includes in the header file. As such, when we split this into `.cpp` and `.h` files, we should probably put most of these in the `.cpp` file. Here's a good StackOverFlow answer explaining why: https://stackoverflow.com/questions/2297567/where-should-include-be-put-in-c